test/e2e/clusterctl_upgrade.go

```diff
@@ -334,6 +336,18 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 
 		By("THE MANAGEMENT CLUSTER WITH THE OLDER VERSION OF PROVIDERS IS UP&RUNNING!")
 
+		machineCRD := &apiextensionsv1.CustomResourceDefinition{}
+		if err := managementClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "machines.cluster.x-k8s.io"}, machineCRD); err != nil {
+			fmt.Printf("Failed to get machine CRD %s: %v\n", machineCRD.Name, err)
+		}
+
+		// Build GroupVersionKind
+		gvk := schema.GroupVersionKind{
+			Group:   machineCRD.Spec.Group,
+			Version: machineCRD.Spec.Versions[0].Name,
+			Kind:    machineCRD.Spec.Names.Kind,
```

Comment:
Please use listKind. This only works because CR always appends the List suffix if it doesn't exist. But I don't want to rely on implicit behavior like this and pass in a basically wrong kind